pkg/asset/machines/master_test.go

```diff
@@ -453,3 +454,90 @@ func TestFencingCredentialsSecretsFile(t *testing.T) {
 		})
 	}
 }
+
+func TestSupportedSNOPlatforms(t *testing.T) {
+	cases := []struct {
+		name                string
+		platform            types.Platform
+		bootstrapInPlace    *types.BootstrapInPlace
+		machinePoolPlatform types.MachinePoolPlatform
+		expectedError       bool
+	}{
+		{
+			name: "AWS platform",
+			platform: types.Platform{
+				AWS: &awstypes.Platform{
+					Region: "us-east-1",
+					DefaultMachinePlatform: &awstypes.MachinePool{
+						InstanceType: "TEST_INSTANCE_TYPE",
+					},
+				},
+			},
+			bootstrapInPlace: &types.BootstrapInPlace{},
+			machinePoolPlatform: types.MachinePoolPlatform{
+				AWS: &awstypes.MachinePool{
+					Zones: []string{"us-east-1a"},
+				},
+			},
+			expectedError: false,
+		},
+		{
+			name: "None platform",
+			platform: types.Platform{
+				None: &none.Platform{},
+			},
+			machinePoolPlatform: types.MachinePoolPlatform{},
+			expectedError:       false,
+		},
+		{
+			name: "IBMCloud platform",
+			platform: types.Platform{
+				IBMCloud: &ibmcloudtypes.Platform{},
+			},
+			bootstrapInPlace: &types.BootstrapInPlace{},
+			machinePoolPlatform: types.MachinePoolPlatform{
+				IBMCloud: &ibmcloudtypes.MachinePool{},
+			},
+			expectedError: false,
+		},
+	}
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			parents := asset.Parents{}
+			parents.Add(
+				&installconfig.ClusterID{
+					UUID:    "test-uuid",
+					InfraID: "test-infra-id",
+				},
+				installconfig.MakeAsset(
+					&types.InstallConfig{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: "test-cluster",
+						},
+						SSHKey:     "ssh-rsa: dummy-key",
+						BaseDomain: "test-domain",
+						Platform:   tc.platform,
+						ControlPlane: &types.MachinePool{
+							Hyperthreading: types.HyperthreadingEnabled,
+							Replicas:       pointer.Int64Ptr(1),
+							Platform:       tc.machinePoolPlatform,
+						},
+					}),
+
+				rhcos.MakeAsset("test-image"),
+				(*rhcos.Release)(pointer.StringPtr("412.86.202208101040-0")),
+				&machine.Master{
+					File: &asset.File{
+						Filename: "master-ignition",
+						Data:     []byte("test-ignition"),
+					},
+				},
+			)
+			master := &Master{}
+			err := master.Generate(context.Background(), parents)
+			if tc.expectedError {
+				assert.Error(t, err)
+			}
```

Comment:
We should also assert that there is no error if none is expected?

```go
if tc.expectedError {
  assert.Error(t, err)
} else {
  assert.NoError(t, err)
}
```

With this check, we do see that in the case of ibmcloud, the unit test fail (probably need to mock):

```
  --- FAIL: TestSupportedSNOPlatforms/IBMCloud_platform#01 (0.00s)
        master_test.go:570: 
            	Error Trace:	/home/thvo/workspace/installer/pkg/asset/machines/master_test.go:570
            	Error:      	Received unexpected error:
            	            	failed generating authenticator for global catalog api: Exactly one of ApiKey or RefreshToken must be specified.
            	            	failed to load IBM SDK services
```

Reply:
I think we can define a zone for the machine pool, that is:

```go
       {
			name: "IBMCloud platform",
			platform: types.Platform{
				IBMCloud: &ibmcloudtypes.Platform{},
			},
			bootstrapInPlace: &types.BootstrapInPlace{},
			machinePoolPlatform: types.MachinePoolPlatform{
				IBMCloud: &ibmcloudtypes.MachinePool{
					Zones: []string{"us-east-1"}, // <--- Here
				},
			},
			expectedError: false,
		},
```